### PR TITLE
GH-2829 Add Sentry on frontend

### LINKED ISSUE #2829: Add Sentry on Frontend
We want to add Sentry on the newly created ErrorBoundaries and also in our logError util. 

We should verify that every onError in the app (usually Apollo), uses this logError.

### PR DESCRIPTION
**Pull request by i-am-chitti**
## Description
This PR adds sentry on frontend. It initializes the Sentry in `ExceptionHandlerProvider` on basis of a check `isSentryInitialized` just to ensure that sentry is not initialized again or have any multiple replays while working in development mode (Hot module replacement).

### Issue
- Closes #2829 

### Screenshots

image

### Note
- Please have a cautious review of options provided during Sentry initialization.

### CODE REVIEW

**Review by charlesBochet — COMMENTED** [review]
Good job! Almost there!
I have left comments :)

Regarding your question, as you have added a package, you need to do `yarn`. It will update the yarn.lock at the root. I've done it here so you don't need to do it anymore

File: packages/twenty-front/src/modules/error-handler/components/ExceptionHandlerProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import { sentryState } from '@/error-handler/states/sentryState';
+import { REACT_APP_SERVER_BASE_URL } from '~/config';
+
+export const ExceptionHandlerProvider: React.FC<React.PropsWithChildren> = ({
+  children,
+}) => {
+  const sentry = useRecoilValue(sentryState);
+  const [isSentryInitialized, setIsSentryInitialized] = useState(false);
+
+  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (charlesBochet) [reply packages/twenty-front/src/modules/error-handler/components/ExceptionHandlerProvider.tsx:14]: This way SentryInitEffect will-rerender but we don't care, it won't impact ExceptionHandlerProvider

**Comment by charlesBochet** [comment]
Also, don't forget to update `packages/twenty-front/src/modules/client-config/graphql/queries/getClientConfig.ts` 
And to run: `yarn nx graphql:data:generate twenty-front` ==> this will update: `packages/twenty-front/src/generated/graphql.tsx` properly (it seems that you have manually edited it, which should not be done as this file is auto-generated)

File: packages/twenty-front/src/modules/error-handler/components/ExceptionHandlerProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from 'react';
```
[the remark this answers is not in the document]
Comment (i-am-chitti) [reply packages/twenty-front/src/modules/error-handler/components/ExceptionHandlerProvider.tsx:1]: @charlesBochet The folder contains components for error boundary, generic error fallback - https://github.com/twentyhq/twenty/tree/main/packages/twenty-front/src/modules/error-handler/components

We can create another folder for exception-handler to have separate concerns.

[the remark this answers is not in the document]
Comment (charlesBochet) [reply packages/twenty-front/src/modules/error-handler/components/ExceptionHandlerProvider.tsx:1]: ok! let's keep error-handler naming for now. We will re-discuss the naming with the team later :)

File: packages/twenty-front/src/index.tsx [hunk 3]
```diff
@@ -36,31 +37,33 @@ root.render(
...
-                          </SnackBarProvider>
-                        </AppThemeProvider>
-                        <PageChangeEffect />
-                      </ObjectMetadataItemsProvider>
-                    </ApolloMetadataClientProvider>
-                  </UserProvider>
-                </ClientConfigProvider>
-              </HelmetProvider>
-            </ApolloProvider>
+            <ExceptionHandlerProvider>
```
[the remark this answers is not in the document]
Comment (i-am-chitti) [reply packages/twenty-front/src/index.tsx:40]: @charlesBochet Anything to address here?

File: packages/twenty-front/src/modules/error-handler/states/sentryState.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { atom } from 'recoil';
+
+import { Sentry } from '~/generated/graphql';
+
+export const sentryState = atom<Sentry | null>({
```
[the remark this answers is not in the document]
Comment (i-am-chitti) [reply packages/twenty-front/src/modules/error-handler/states/sentryState.ts]: Moved it to same folder where client-config states are being managed.

File: packages/twenty-front/src/modules/error-handler/components/ExceptionHandlerProvider.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (i-am-chitti) [reply packages/twenty-front/src/modules/error-handler/components/ExceptionHandlerProvider.tsx]: Thanks @charlesBochet for this amazing approach. I wasn't aware of this pattern. Have addressed this.

**Comment by i-am-chitti** [comment]
@charlesBochet This is ready for a re-review. Have updated the GQL generated data via the command - `yarn nx graphql:data:generate twenty-front`.

After doing so, some unwanted changes (not related to this feature) have popped up in `graphql.tsx`. 
image

**Comment by charlesBochet** [comment]
@i-am-chitti I'm taking a look in a few mins

File: packages/twenty-front/src/index.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (charlesBochet) [reply packages/twenty-front/src/index.tsx:40]: nope, all good!

**Review by charlesBochet — APPROVED** [review]
Looks good, thank you!

**Pull request merged**